bot/modules/bartek/youtube_handler.py

```diff
@@ -0,0 +1,46 @@
+import aiohttp
+import logging
+import re
+import json
+
+from .text_handler import TextHandler
+
+
+class YoutubeHandler:
+    def __init__(
+        self,
+        log: logging.Logger,
+        textHandler: TextHandler,
+        session: aiohttp.ClientSession,
+    ) -> None:
+        self.textHandler: TextHandler = textHandler
+        self.log: logging.Logger = log
+        self.session: aiohttp.ClientSession = session
+
+    async def isOffending(self, url: str) -> bool:
+        try:
+            async with self.session.get(url) as resp:
+                yt_page_txt: str = await resp.text()
+        except:
+            self.log.error("Bad url - youtube")
+            return False
+
+        # pattern to hopefully recieve json with wanted data
+        re_json: re.Pattern[str] = re.compile(r">var ytInitialData = (\{.+?\});<")
+        json_data: object = json.loads(re_json.search(yt_page_txt).group(1))
```

Comment:
This will raise a `TypeError` if the regex search fails. Please handle exceptions and also possibly split one-liners like this for readability.